src/main/java/com/github/minecraftschurlimods/arsmagicalegacy/client/gui/inscriptiontable/InscriptionTableScreen.java

```diff
@@ -0,0 +1,207 @@
+package com.github.minecraftschurlimods.arsmagicalegacy.client.gui.inscriptiontable;
+
+import com.github.minecraftschurlimods.arsmagicalegacy.api.ArsMagicaAPI;
+import com.github.minecraftschurlimods.arsmagicalegacy.api.spell.ISpell;
+import com.github.minecraftschurlimods.arsmagicalegacy.api.spell.ISpellItem;
+import com.github.minecraftschurlimods.arsmagicalegacy.api.spell.ISpellPart;
+import com.github.minecraftschurlimods.arsmagicalegacy.api.spell.ShapeGroup;
+import com.github.minecraftschurlimods.arsmagicalegacy.client.gui.SelfClearingEditBox;
+import com.github.minecraftschurlimods.arsmagicalegacy.client.gui.dragndrop.DragArea;
+import com.github.minecraftschurlimods.arsmagicalegacy.common.block.inscriptiontable.InscriptionTableMenu;
+import com.github.minecraftschurlimods.arsmagicalegacy.common.util.TranslationConstants;
+import com.mojang.blaze3d.platform.InputConstants;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.components.EditBox;
+import net.minecraft.client.gui.components.events.GuiEventListener;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.player.Inventory;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class InscriptionTableScreen extends AbstractContainerScreen<InscriptionTableMenu> {
+    private static final ResourceLocation GUI = new ResourceLocation(ArsMagicaAPI.MOD_ID, "textures/gui/inscription_table.png");
+    private static final Component SEARCH_LABEL = Component.translatable(TranslationConstants.INSCRIPTION_TABLE_SEARCH);
+    private final List<DragArea<SpellPartDraggable>> dragAreas = new ArrayList<>();
+    private SpellPartDraggable dragged;
+    private EditBox searchBar;
+    private SpellPartSourceArea sourceArea;
+    private SpellGrammarArea spellGrammarArea;
+    private ShapeGroupListArea shapeGroupArea;
+
+    public InscriptionTableScreen(InscriptionTableMenu pMenu, Inventory pPlayerInventory, Component pTitle) {
+        super(pMenu, pPlayerInventory, pTitle);
+        imageWidth = 220;
+        imageHeight = 252;
+    }
+
+    public static void onSlotChanged() {
+        if (Minecraft.getInstance().screen instanceof InscriptionTableScreen screen) {
+            screen.sync();
+            if (screen.menu.getSlot(0).getItem().getItem() instanceof ISpellItem) {
+                screen.menu.getSpellRecipe().ifPresent(screen::existingRecipe);
+            }
+        }
+    }
+
+    @Override
+    protected void init() {
```

Comment:
Why do we no longer have the ability to name the spell recipe?

Reply:
Because there is no point, and it clutters the GUI. The spell name needs to be set again later anyway (the crafting altar doesn't transfer the spell recipe name onto the spell)

Reply:
In what way is there no point? the name is to differentiate recipes without having to open the recipe

Reply:
People tend to craft the spell immediately after making the recipe (unless they are missing the items). I don't see the point of having different names for what is effectively a one time use item.
But you know what? We could ask the community on how they feel about this.

Reply:
Yes but I keep my recipes after creating the recipe and I don't throw it away after making the spell (use case: you die and need to recreate your spells). And I don't want to have to rename the recipes in an anvil.

Reply:
That is a fair point. Though others might think they just take up space not needed.
I have made a poll on the Discord.

Reply:
The name bar has been readded.